src/Core/Form/IdentifiableObject/DataProvider/ProductFormDataProvider.php

```diff
@@ -642,36 +642,41 @@ private function presentTags(array $localizedTagsList): array
     /**
      * @param ProductForEditing $productForEditing
      *
-     * @return array<string, int|array<int, int|array<string, string|int>>>
+     * @return array<string, array<int|string, array<int|string, int|string|null>|int>>
```

Comment:
Do you think it's welcome to create a DTO that replicate this structure to simplify ?

Reply:
it will be also easier to access the data with a getter than list of keys.